ocpmodels/trainers/base_trainer.py

```diff
@@ -403,24 +403,25 @@ def load_checkpoint(self, checkpoint_path):
         self.best_val_metric = checkpoint.get("best_val_metric", None)
         self.primary_metric = checkpoint.get("primary_metric", None)
 
-        # Load model, optimizer, normalizer state dict.
-        # if trained with ddp and want to load in non-ddp, modify keys from
-        # module.module.. -> module..
-        first_key = next(iter(checkpoint["state_dict"]))
-        if (
-            not distutils.initialized() or self.config["noddp"]
-        ) and first_key.split(".")[1] == "module":
-            # No need for OrderedDict since dictionaries are technically ordered
-            # since Python 3.6 and officially ordered since Python 3.7
-            new_dict = {k[7:]: v for k, v in checkpoint["state_dict"].items()}
-            self.model.load_state_dict(new_dict)
-        elif distutils.initialized() and first_key.split(".")[1] != "module":
+        ckpt_key_count = next(iter(checkpoint["state_dict"])).count("module")
+        mod_key_count = next(iter(self.model.state_dict())).count("module")
+        key_count_diff = mod_key_count - ckpt_key_count
```

Comment:
Trying to understand the changes — in which situations did the earlier checkpoint loading logic fail?

Might be worth adding comments in the code to explain how `key_count_diff` being == 0 or > 0 or < 0 map to training or running inference with or without DDP and such.

Reply:
In principle there are total of 9 combinations we will have to take care of. Not having any "module." (for CPU or without use of OCPDataParallel), 1 "module." on single GPU, 2 "module." for distributed training and their corresponding checkpoints (3*3 possibilities). We didn't have no "module." related logic earlier. Although, that's a very niche use case, the current logic should take care of all possibilities.